src/internal_events/http.rs

```diff
@@ -86,15 +125,22 @@ pub struct HttpDecompressError<'a> {
 
 impl<'a> InternalEvent for HttpDecompressError<'a> {
     fn emit_logs(&self) {
-        warn!(
+        error!(
             message = "Failed decompressing payload.",
             encoding= %self.encoding,
             error = %self.error,
+            stage = "receiving",
             internal_log_rate_secs = 10
         );
     }
 
     fn emit_metrics(&self) {
         counter!("parse_errors_total", 1);
+        counter!(
+            "component_errors_total", 1,
+            "error_type" => "parse_failed",
+            "stage" => "receiving",
```

Comment:
Does stage refer to when in the lifecycle of the source the log/metric is emitted?

Reply:
Correct.